Tests/Realm.Tests/Sync/SyncTestHelpers.cs

```diff
@@ -75,34 +97,42 @@ public static string[] ExtractBaasSettings(string[] args)
         {
             var result = new List<string>();
 
-            string baasUrl = null;
-            string baasAppId = null;
-
             for (var i = 0; i < args.Length; i++)
             {
                 switch (args[i])
                 {
                     case "--baasurl":
-                        baasUrl = args[++i];
-                        break;
-                    case "--baasappid":
-                        baasAppId = args[++i];
+                        _baseUri = new Uri(args[++i]);
                         break;
                     default:
                         result.Add(args[i]);
                         break;
                 }
             }
 
-            if (baasUrl != null && baasAppId != null)
+            ExtractBaasApps();
+
+            return result.ToArray();
+        }
+
+        private static void ExtractBaasApps()
```

Comment:
This gets called either during initial startup (if run via the command line) or the first time a baas test is run. It's synchronous because I didn't want to refactor a whole lot of code around it to make it async and since it's only called once per process, it's not a huge deal.